modules/dmrpp_module/CurlHandlePool.cc

```diff
@@ -395,23 +405,26 @@ void CurlHandlePool::release_handle(dmrpp_easy_handle *handle) {
     std::lock_guard<std::recursive_mutex> lock_me(d_get_easy_handle_mutex);
 
     // TODO Add a call to curl reset() here. jhrg 9/23/20
-
-#if KEEP_ALIVE
-    handle->d_url = nullptr;
-    handle->d_chunk = nullptr;
-    handle->d_in_use = false;
-#else
-    // This is to test the effect of libcurl Keep Alive support
-    // Find the handle; erase from the vector; delete; allocate a new handle and push it back on
-    for (std::vector<dmrpp_easy_handle *>::iterator i = d_easy_handles.begin(), e = d_easy_handles.end(); i != e; ++i) {
-        if (*i == handle) {
-            BESDEBUG("dmrpp:5", "Found a handle match for the " << i - d_easy_handles.begin() << "th easy handle." << endl);
-            delete handle;
-            *i = new dmrpp_easy_handle();
-            break;
+    //   I stuck it in the "replace" block below. ndp 08/07/23
+
+    if(replace || !KEEP_ALIVE) {
```

Comment:
I think we should talk about the KEEP_ALIVE compile-time constant, since it could make the 'replace' parameter meaningless (if KEEP_ALIVE is 0, then the if is always true no matter what replace is).